### PR TITLE
Stop injecting compiler arguments via PMCS

### PR DESCRIPTION
**Pull request by chamons**
- Part of the PMCS removal work. 
- A second PR will later remove classic XI the associated PMCS code, like XM
- Requires https://github.com/xamarin/macios-binaries/pull/2 to land into master
- apidiff comes back clean locally, but please review very closely.

### CODE REVIEW

File: src/Makefile [hunk 1]
```diff
@@ -48,6 +48,9 @@ COMMON_TARGET_DIRS = \
 	build/common             \
 	build/common/NativeTypes \
 
+32_ARGS = -define:ARCH_32
+64_ARGS = -define:ARCH_64
```
Comment (spouliot) [inline src/Makefile]: minor: generally we put numbers at the end, like `ARGS_32`

File: src/Makefile [hunk 2]
```diff
@@ -65,7 +68,9 @@ IOS_CORE_SOURCES += $(IOS_EXTRA_SOURCES)
 IOS_SOURCES += $(IOS_EXTRA_SOURCES)
 
 IOS_GENERATOR_FLAGS = -inline-selectors -d:IOS -process-enums -warnaserror:$(IOS_GENERATOR_WARNASERROR)
+IOS_GENERATOR_native_FLAGS = -d:XAMCORE_2_0 -d:__UNIFIED__
```
Comment (spouliot) [inline src/Makefile]: minor: `native` in caps - but if I understand correctly where you're going this is temporary :)

File: src/Makefile [hunk 3]
```diff
@@ -439,8 +454,11 @@ MAC_WARNINGS_TO_FIX := 114,108
 # 4014 is "The statement is not awaited and execution of current method continues before the call is completed. Consider using `await' operator or calling `Wait' method"
 MAC_WARNINGS_TO_FIX := $(MAC_WARNINGS_TO_FIX),4014
 
-MAC_COMMON_DEFINES = -define:NET_2_0,XAMARIN_MAC,MONOMAC
+MAC_COMMON_DEFINES = -define:NET_2_0,XAMARIN_MAC,MONOMAC,XAMCORE_2_0,__UNIFIED__
+MAC_full_ARGS = -sdk:4.5 -define:NO_SYSTEM_DRAWING -define:XAMMAC_SYSTEM_MONO
+MAC_mobile_ARGS = -nostdlib -r:mscorlib -lib:$(abspath $(MAC_DESTDIR)/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac)
```
Comment (spouliot) [inline src/Makefile]: minor: full, mobile -> caps
might want to use `modern` ?

File: src/Makefile (hunk 2 above)
Comment (chamons) [reply src/Makefile]: This native I believe has to match the native we pass in here:

https://github.com/xamarin/xamarin-macios/blob/master/src/Makefile#L158

File: src/Makefile (hunk 3 above)
Comment (chamons) [reply src/Makefile]: Same here, full and mobile need to match these:

https://github.com/xamarin/xamarin-macios/blob/master/src/Makefile#L549

and yes, doing a scrub of mobile => full has been on my todo list for some point. We have a bunch of places though. :(

File: src/Makefile (hunk 1 above)
Comment (chamons) [reply src/Makefile]: agreed

**Comment by rolfbjarne** [comment]
@chamons did you check if the generated sources changed?

**Comment by chamons** [comment]
@rolfbjarne No, but that'd be a great thing to check. Let me do that.

**Comment by chamons** [comment]
@rolfbjarne 17409 files sames, 96 different. All of them are binaries except a few generated constants dealing with hashes and branch names.

👍 fwict.

**Comment by chamons** [comment]
Once we get solid test results this is ready to land I believe.

**Comment by chamons** [comment]
```21:39:59 Validated file permissions for Xamarin.Mac.
21:39:59 There are files with invalid permissions (all installed files at least be readable by everybody, and writable by owner: 0644): 
21:39:59 -rw-------  1 builder  staff  4872169 Nov 28 16:38 /Users/builder/jenkins/workspace/xamarin-macios-pr-builder/_ios-build/Library/Frameworks/Xamarin.iOS.framework/Versions/git/lib/mono/2.1/monotouch.dll.mdb```

Hmm. Let me try make check-permissions locally...

**Comment by chamons** [comment]
What in the world. It is failing locally. Maybe PMCS is doing something w\ permissions. I'll have to dig further.

**Comment by chamons** [comment]
Let's see if 022336abd9b2e87195bcfd70a30c67dcdc2346db fixes it on bots as it does locally.

**Comment by chamons** [comment]
```***** NSLayoutManager_DrawGlyphsForGlyphRange
2017-12-05 13:36:12.971 apitest[57432:1014908] !!! _NSGlyphTreeNotShownAttributeForGlyphAtIndex invalid glyph index 0
2017-12-05 13:36:12.977 apitest[57432:1014908] !!! _NSLayoutTreeLineFragmentRectForGlyphAtIndex invalid glyph index 0
2017-12-05 13:36:12.977 apitest[57432:1014908] !!! _NSLayoutTreeLineFragmentRectForGlyphAtIndex invalid glyph index 0
2017-12-05 13:36:12.977 apitest[57432:1014908] !!! _NSLayoutTreeRunLocationForGlyphAtIndex invalid glyph index 0
2017-12-05 13:36:12.977 apitest[57432:1014908] !!! _NSLayoutTreeRunLocationForGlyphAtIndex invalid glyph index 0
2017-12-05 13:36:12.977 apitest[57432:1014908] !!! _NSGlyphTreeNotShownAttributeForGlyphAtIndex invalid glyph index 0
2017-12-05 13:36:12.977 apitest[57432:1014908] !!! _NSLayoutTreeLineFragmentRectForGlyphAtIndex invalid glyph index 0
2017-12-05 13:36:12.977 apitest[57432:1014908] !!! _NSLayoutTreeLineFragmentRectForGlyphAtIndex invalid glyph index 0
2017-12-05 13:36:12.977 apitest[57432:1014908] !!! _NSLayoutTreeRunLocationForGlyphAtIndex invalid glyph index 0
2017-12-05 13:36:12.978 apitest[57432:1014908] !!! _NSLayoutTreeRunLocationForGlyphAtIndex invalid glyph index 0
2017-12-05 13:36:12.978 apitest[57432:1014908] !!! _NSGlyphTreeNotShownAttributeForGlyphAtIndex invalid glyph index 0
2017-12-05 13:36:12.978 apitest[57432:1014908] !!! _NSLayoutTreeLineFragmentRectForGlyphAtIndex invalid glyph index 0
2017-12-05 13:36:12.978 apitest[57432:1014908] !!! _NSLayoutTreeLineFragmentRectForGlyphAtIndex invalid glyph index 0
2017-12-05 13:36:12.978 apitest[57432:1014908] !!! _NSLayoutTreeRunLocationForGlyphAtIndex invalid glyph index 0
2017-12-05 13:36:12.978 apitest[57432:1014908] !!! _NSLayoutTreeRunLocationForGlyphAtIndex invalid glyph index 0
2017-12-05 13:36:12.978 apitest[57432:1014908] !!! _NSGlyphTreeNotShownAttributeForGlyphAtIndex invalid glyph index 0
2017-12-05 13:36:12.978 apitest[57432:1014908] !!! _NSLayoutTreeLineFragmentRectForGlyphAtIndex invalid glyph index 0
2017-12-05 13:36:12.978 apitest[57432:1014908] !!! _NSLayoutTreeLineFragmentRectForGlyphAtIndex invalid glyph index 0
2017-12-05 13:36:12.978 apitest[57432:1014908] !!! _NSLayoutTreeRunLocationForGlyphAtIndex invalid glyph index 0
2017-12-05 13:36:12.978 apitest[57432:1014908] !!! _NSLayoutTreeRunLocationForGlyphAtIndex invalid glyph index 0
2017-12-05 13:36:12.978 apitest[57432:1014908] !!! Suppressing further NSLayoutTree logs
2017-12-05 13:36:12.978 apitest[57432:1014908] !!! _NSGlyphTreeNotShownAttributeForGlyphAtIndex invalid glyph index 0
2017-12-05 13:36:12.978 apitest[57432:1014908] !!! _NSGlyphTreeNotShownAttributeForGlyphAtIndex invalid glyph index 0
2017-12-05 13:36:12.978 apitest[57432:1014908] !!! _NSGlyphTreeNotShownAttributeForGlyphAtIndex invalid glyph index 0
2017-12-05 13:36:12.978 apitest[57432:1014908] !!! _NSGlyphTreeNotShownAttributeForGlyphAtIndex invalid glyph index 0
2017-12-05 13:36:12.978 apitest[57432:1014908] !!! _NSGlyphTreeNotShownAttributeForGlyphAtIndex invalid glyph index 0
2017-12-05 13:36:12.978 apitest[57432:1014908] !!! _NSGlyphTreeNotShownAttributeForGlyphAtIndex invalid glyph index 0
2017-12-05 13:36:12.978 apitest[57432:1014908] !!! _NSGlyphTreeNotShownAttributeForGlyphAtIndex invalid glyph index 0
2017-12-05 13:36:12.978 apitest[57432:1014908] !!! _NSGlyphTreeNotShownAttributeForGlyphAtIndex invalid glyph index 0
2017-12-05 13:36:12.978 apitest[57432:1014908] !!! _NSGlyphTreeNotShownAttributeForGlyphAtIndex invalid glyph index 0
2017-12-05 13:36:12.978 apitest[57432:1014908] !!! _NSGlyphTreeNotShownAttributeForGlyphAtIndex invalid glyph index 0
2017-12-05 13:36:12.978 apitest[57432:1014908] !!! _NSGlyphTreeNotShownAttributeForGlyphAtIndex invalid glyph index 0
2017-12-05 13:36:12.979 apitest[57432:1014908] !!! _NSGlyphTreeNotShownAttributeForGlyphAtIndex invalid glyph index 0
2017-12-05 13:36:12.979 apitest[57432:1014908] !!! Suppressing further NSGlyphTree logs
apitest(57432,0xa6f4a1c0) malloc: *** mach_vm_map(size=8388608) failed (error code=3)
*** error: can't allocate region securely
*** set a breakpoint in malloc_error_break to debug```

Interesting failure in some NSGlyph test code that seems totally unrelated. I'll request another build when our bots are online again.

**Comment by chamons** [comment]
build

**Comment by chamons** [comment]
Test failure - https://github.com/xamarin/maccore/issues/592

**Comment by spouliot** [comment]
xtro failure is unrelated to this PR and will be fixed separately

**Comment by chamons** [comment]
```
23:21:22 	at org.jenkinsci.remoting.protocol.NetworkLayer.onRecvClosed(NetworkLayer.java:154)
23:21:22 	at org.jenkinsci.remoting.protocol.impl.NIONetworkLayer.ready(NIONetworkLayer.java:179)
23:21:22 	at org.jenkinsci.remoting.protocol.IOHub$OnReady.run(IOHub.java:721)
```

Network issues.

**Comment by chamons** [comment]
build

**Comment by spouliot** [comment]
above xtro failure fixed with PR #3131

**Pull request merged**